### PR TITLE
bash helpers

### PR DESCRIPTION
**Pull request by tekakutli**
in particular, a simple script that allows you to transcribe what you are hearing
don't bother trying to cut video segments, or transcribe long audios, instead do it on the fly from your speakers, one keybinding away
dependencies: linux, pipewire, ffmpeg, whisper

### CODE REVIEW

File: experiments/EXAMPLE_gpt_raw.yml [hunk 1]
```diff
@@ -0,0 +1,144 @@
+name: CHANGEME_your_experiment_name
```
Comment (152334H) [inline experiments/EXAMPLE_gpt_raw.yml:1]: could you change your script to run on a different yml filename, and rename this file back to `EXAMPLE_gpt.yml` instead? The colab notebook is reliant on the `EXAMPLE_gpt.yml` file in a lot of places

File: speakers_stt.sh [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+LIVE_RECORD="./recording.wav"
+WAV_RECORD="./recording_whisper.wav"
+pw-record $LIVE_RECORD
+ffmpeg -y -i "$LIVE_RECORD" -ar 16000 -ac 1 -c:a pcm_s16le $WAV_RECORD
+
+
+
+# TINY IS FAST AND ENOUGH
+if [[ "$LANG_FROM" == "en" ]]; then
+    $PATH_TO_WHISPER/main -m $PATH_TO_MODELS/ggml-tiny.bin -l en -f "$WAV_RECORD" -otxt
```
Comment (152334H) [inline speakers_stt.sh:52]: Are you aware that you are not using the English-specific model here? `ggml-tiny.en.bin` and `ggml-tiny.bin` are different models. The former would perform much better than the latter.

File: helper-bash.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env sh
+
+#source this in your profile
+
+TORTOISE_DIR="/home/$USER/code/tortoise-tts-fast/"
+TORTRAIN_DIR="/home/$USER/code/DL-Art-School/"
+TTS_TEXT="or a given problem should be approached using traditional or conventional machine learning or should I be doing deep learning."
+EXPERIMENT_NAME="dade"
```
Comment (152334H) [inline helper-bash.sh:8]: [1]

File: helper-bash.sh [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+     python get_conditioning_latents.py --voice $1
+     cd $($TORTOISE_DIR)tortoise/voices/
+     mkdir $2
+     cd $2
+     cp $($TORTOISE_DIR)results/conditioning_latents/$1.pth ./$2.pth
+}
+
+alias example="--voice emma --seed 42 --text '$TTS_TEXT'"
+alias tortoise="cd $TORTOISE_DIR'' && python tortoise/do_tts.py --kv_cache --half --preset very_fast"
+
```
Comment (152334H) [inline helper-bash.sh:21]: [1]

File: quick-experiment.sh [hunk 5]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+#
+# quick access to ./experiments/EXAMPLE_gpt.yml most important settings
+
+NAME_EXPERIMENT="dud"
+NAME_DATASET="dud-dataset"
+PATH_TRAINING="/tmp/train/metadata.csv"
+NAME_VALIDATION="dud-validation"
+PATH_VALIDATION="/tmp/val/metadata.csv"
+
```
Comment (152334H) [inline quick-experiment.sh:10]: [1]

File: speakers_stt.sh [hunk 6]
```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env sh
+
+# Transcribe what you're hearing, one keybinding away
+# Put something like this in your sway/config:
+# bindsym $mod+Shift+return exec alacritty -e bash /home/tekakutli/code/DL-Art-School/speakers_stt.sh
+
```
Comment (152334H) [inline speakers_stt.sh:6]: [1]

File: speakers_stt.sh [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+#THIS ASUMES YOU ALREADY HAVE WHISPER.CPP INSTALLED, SOMEWHERE
+#https://github.com/ggerganov/whisper.cpp
+#SET THESE WITH YOURS
+LANG_FROM="en"
+PATH_TO_WHISPER="/home/$USER/code/whisper.cpp"
+PATH_TO_MODELS="/home/$USER/files/models"
+OFFSET=0
+PATH_TRAINING="/tmp/val/"
+
```
Comment (152334H) [inline speakers_stt.sh:19]: [1]

**Review by 152334H — CHANGES_REQUESTED** [review]
All places where I see variables that may need to be manually adjusted by the user have been marked with [1].

As you may notice, there are quite a few locations marked with [1]. Because the complexity of configuring this script seems likely to be above the capabilities of the average user, I do not think it is a good idea to add this script to the repository unless it becomes more user-friendly && safe to input errors.

If you are reluctant to make such changes (which I expect), you could also maintain these scripts on a separate gist/repo, and I could add a link + short blub to them on the README.md.

**Comment by tekakutli** [comment]
I still think that the speaker_stt.sh script is particularly useful, but maybe you are right and it's more like something for a gist

**Comment by 152334H** [comment]
It's definitely useful and cool that you can record audio and make parallel whisper transcripts with single button push!

But I am afraid that in it's current state I would get a lot of user issues for script usage and documentation, and I would not want to address that myself.

**Comment by tekakutli** [comment]
I created a separate repo https://github.com/tekakutli/tortoise-scripts

**Pull request closed without merging**